components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtilTest.java

```diff
@@ -18,18 +18,84 @@
 
 package org.wso2.carbon.identity.sso.saml.util;
 
-import static org.testng.Assert.assertFalse;
-import static org.testng.Assert.assertTrue;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.base.IdentityException;
 
+import org.opensaml.xml.XMLObject;
+
+import java.io.IOException;
+
+import static org.testng.Assert.*;
+
+/**
+ * Unit test cases for SAMLSSOUtil.
+ */
 public class SAMLSSOUtilTest {
-    @org.testng.annotations.Test
+
+    @Test
+    public void testUnmarshall() throws Exception {
+        XMLObject xmlObject = SAMLSSOUtil.unmarshall(TestConstants.DECODED_POST_AUTHN_REQUEST);
+        assertEquals(xmlObject.getDOM().getAttributeNode("Destination").getValue(),
+                "https://localhost:9443/samlsso");
+        assertEquals(xmlObject.getDOM().getAttributeNode("Reason").getValue(), "Single Logout");
+    }
+
+    @Test
+    public void testMarshall() throws Exception {
+        assertEquals(SAMLSSOUtil.marshall(SAMLSSOUtil.unmarshall(TestConstants.DECODED_POST_AUTHN_REQUEST)),
+                TestConstants.DECODED_POST_AUTHN_REQUEST);
+    }
+
+    @Test
+    public void testEncode() throws Exception {
+
+        String encodedXml = SAMLSSOUtil.encode(TestConstants.AUTHN_FAILED_SAML_RESPONSE);
+        assertFalse(encodedXml.isEmpty(), "Encoded xml should not be empty.");
+        assertFalse(encodedXml.contains(" "), "Encoded xml should not contain spaces.");
+    }
+
+    @Test(expectedExceptions = IdentityException.class)
+    public void testDecode() throws Exception {
+
+        assertEquals(SAMLSSOUtil.decode(TestConstants.ENCODED_REDIRECT_AUTHN_REQUEST),
+                TestConstants.DECODED_REDIRECT_AUTHN_REQUEST);
+        SAMLSSOUtil.decode(TestConstants.GENERAL_STRING);
+    }
+
+    @Test
+    public void testDecodeForPost() throws Exception {
+
+        assertEquals(SAMLSSOUtil.decodeForPost(TestConstants.ENCODED_POST_AUTHN_REQUEST),
+                TestConstants.DECODED_POST_AUTHN_REQUEST);
+    }
+
+    @Test
+    public void testCompressResponse() throws Exception {
+
+        try {
+            assertFalse(SAMLSSOUtil.compressResponse(TestConstants.AUTHN_FAILED_SAML_RESPONSE).isEmpty(),
+                    "Compressed Authentication Failed SAML response should not be empty.");
+        } catch (IOException ex) {
+            fail("Error while compressing Authentication Failed SAML response.");
```

Comment:
Dont catch exceptions. Instead define them as expected exceptions.

Also you have written two tests within same test method here. Please write them as separate tests

Reply:
here defining them as expected exceptions means, we let the test pass when the exception is thrown.
Instead here what I wanted to do was if the exception is thrown, then should fail the test.